cpu/arm7_common/arm_cpu.c

```diff
@@ -94,3 +94,47 @@ void thread_print_stack(void)
 
     printf("STACK (%d)= %X \n", i, *s);
 }
+
+void *thread_isr_stack_start(void)
+{
+    extern uintptr_t __stack_irq_start;
+    return (void *)&__stack_irq_start;
+}
+
+void *thread_isr_stack_pointer(void)
+{
+    unsigned _cpsr;
+    void *_sp;
+
+    unsigned state = irq_disable();
+
+    /* store current mode */
+    __asm volatile ("mrs %0, cpsr" : "=r" (_cpsr));
+    /* enter interrupt mode */
+    __asm volatile ("msr cpsr, 0x12");
+    /* read stack pointer */
+    __asm volatile ("mov %0, r13" : "=r" (_sp) );
+    /* restore previous mode */
+    __asm volatile ("msr cpsr, %0" : "=r" (_cpsr));
+
+    irq_restore(state);
+
+    return _sp;
+}
+
+/* This function returns the number of bytes used on the ISR stack */
+int thread_isr_stack_usage(void)
+{
+    extern uintptr_t __stack_irq_start;
+    extern uintptr_t __stack_irq_size;
+
+    uintptr_t *ptr = &__stack_irq_start - (unsigned) &__stack_irq_size;
+
+    while(((*ptr) == STACK_CANARY_WORD) && (ptr < &__stack_irq_start)) {
+        ++ptr;
+    }
```

Comment:
This will only work with `DEVELHELP` defined. (But that is also true for Cortex M.)

The [documentation](http://api.riot-os.org/group__core__thread.html#gabb6763813ca04ff4b87b47f3d524505b) sadly is very poor. Intuitively I would have guessed that this function should return the current ISR stack usage and must be called from interrupt context only. But I confirm that the implementation here should behave exactly as the Cortex M version.